tests/test_date_pyparser.py

```
@@ -3,997 +3,1046 @@
+                 ('YMD', '05/13/07', 2007, 5, 13),
+                 ('MDY', '13/06/07', 2013, 6,  7),
+                 ('DMY', '05/13/07', 2007, 5, 13)]:
+        key, test, y, m, d = case
+        test = before + test + after
+        if key == order:
+            for p in (parser_loose, parser_strict):
+                try:
+                    pairs = p.parse_string(test).as_list()
+                except Exception as e:      # pragma: no cover
+                    assert False, (type(e).__name__ + ' on ' + repr(test)
+                                   + ': ' + str(e))
```

Ruff (0.15.6)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)